packages/x-date-pickers-pro/src/MultiInputDateRangeField/MultiInputDateRangeField.tsx

```diff
@@ -137,63 +138,22 @@ const MultiInputDateRangeField = React.forwardRef(function MultiInputDateRangeFi
     className: classes.separator,
   });
 
-  const {
-    startDate: {
-      onKeyDown: onStartInputKeyDown,
-      ref: startInputRef,
-      readOnly: startReadOnly,
-      inputMode: startInputMode,
-      ...startDateProps
-    },
-    endDate: {
-      onKeyDown: onEndInputKeyDown,
-      ref: endInputRef,
-      readOnly: endReadOnly,
-      inputMode: endInputMode,
-      ...endDateProps
-    },
-  } = useMultiInputDateRangeField<TDate, FieldsTextFieldProps>({
+  const fieldResponse = useMultiInputDateRangeField<TDate, FieldsTextFieldProps>({
     sharedProps: { ...dateFieldInternalProps, disabled },
     startTextFieldProps,
     endTextFieldProps,
     unstableStartFieldRef,
     unstableEndFieldRef,
-    startInputRef: startTextFieldProps.inputRef,
-    endInputRef: endTextFieldProps.inputRef,
   });
 
+  const startDateProps = useConvertFieldResponseIntoMuiTextFieldProps(fieldResponse.startDate);
```

Comment:
Nice helper. WHat about converting it's name to `convertFieldResponseIntoMuiTextFieldProps`. The `use` prefix is reserved for hooks, and this one seems to be a pure function (no state, no effect)

Reply:
You are right, even with the v7 approach it does not have any hook-related logic